### PR TITLE
[Core][Tests] Add attribute tests and decompotion tests for Scaled Dot Product Attention

### PR DESCRIPTION
**Pull request by PiotrKrzem**
### Details:
 - Added attribute tests to ensure the correct attributes are set for the Scaled Dot Product Attention operation.
 - Introduced decomposition tests for Scaled Dot Product Attention to validate the correctness of the decomposition process.

### Tickets:
 - 123773

### CODE REVIEW

**Review by mitruska — COMMENTED** [review]
Please fix the build to run those tests on CI.

File: src/common/transformations/tests/op_conversions/scaled_dot_product_decomposition_test.cpp [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    const std::shared_ptr<ov::Node> scale,
+    const bool casual);
+
+TEST_F(TransformationTestsF, ScaledDotProductAttentionDecompositionStatic) {
+    const PartialShape query_key_value_mask_scale__shape{32, 32, 32};
+
+    const auto query = std::make_shared<ov::op::v0::Parameter>(element::f32, query_key_value_mask_scale__shape);
+    const auto key = std::make_shared<ov::op::v0::Parameter>(element::f32, query_key_value_mask_scale__shape);
+    const auto value = std::make_shared<ov::op::v0::Parameter>(element::f32, query_key_value_mask_scale__shape);
+    const auto attention_mask = std::make_shared<ov::op::v0::Parameter>(element::f32, query_key_value_mask_scale__shape);
```
[the remark this answers is not in the document]
Comment (PiotrKrzem) [reply src/common/transformations/tests/op_conversions/scaled_dot_product_decomposition_test.cpp]: Fixed!

File: src/common/transformations/tests/op_conversions/scaled_dot_product_decomposition_test.cpp [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        model_ref = std::make_shared<ov::Model>(NodeVector{scaled_dot_product_attention},
+                                                ParameterVector{query, key, value, attention_mask, scale});
+    }
+}
+
+TEST_F(TransformationTestsF, ScaledDotProductAttentionDecompositionDynamic) {
+    const PartialShape query_key_value_mask_shape{-1, -1, -1};
+    const PartialShape scale_shape{};
+
+    const auto query = std::make_shared<ov::op::v0::Parameter>(element::f32, query_key_value_mask_scale__shape);
```
[the remark this answers is not in the document]
Comment (PiotrKrzem) [reply src/common/transformations/tests/op_conversions/scaled_dot_product_decomposition_test.cpp]: Applied!

File: src/common/transformations/tests/op_conversions/scaled_dot_product_decomposition_test.cpp [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    const std::shared_ptr<ov::Node> key,
+    const std::shared_ptr<ov::Node> value,
+    const std::shared_ptr<ov::Node> attention_mask,
+    const std::shared_ptr<ov::Node> scale,
+    const bool casual);
+
+TEST_F(TransformationTestsF, ScaledDotProductAttentionDecompositionStatic) {
+    const PartialShape query_key_value_mask_scale__shape{32, 32, 32};
+
+    const auto query = std::make_shared<ov::op::v0::Parameter>(element::f32, query_key_value_mask_scale__shape);
```
[the remark this answers is not in the document]
Comment (PiotrKrzem) [reply src/common/transformations/tests/op_conversions/scaled_dot_product_decomposition_test.cpp]: Applied!

File: src/common/transformations/tests/op_conversions/scaled_dot_product_decomposition_test.cpp [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    const bool casual);
+
+TEST_F(TransformationTestsF, ScaledDotProductAttentionDecompositionStatic) {
+    const PartialShape query_key_value_mask_scale_shape{32, 32, 32};
+
+    const auto query = std::make_shared<ov::op::v0::Parameter>(element::f32, query_key_value_mask_scale_shape);
+    const auto key = std::make_shared<ov::op::v0::Parameter>(element::f32, query_key_value_mask_scale_shape);
+    const auto value = std::make_shared<ov::op::v0::Parameter>(element::f32, query_key_value_mask_scale_shape);
+    const auto attention_mask = std::make_shared<ov::op::v0::Parameter>(element::f32, query_key_value_mask_scale_shape);
+    const auto scale = std::make_shared<ov::op::v0::Parameter>(element::f32, query_key_value_mask_scale_shape);
```
Comment (mmikolajcz) [inline src/common/transformations/tests/op_conversions/scaled_dot_product_decomposition_test.cpp:49]: + other scale inputs - according to specification, scale should be either scalar or 1-element 1d tensor

File: src/common/transformations/tests/op_conversions/scaled_dot_product_decomposition_test.cpp [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+    }
+
+    {
+        const auto scaled_dot_product_attention =
+            scaled_dot_product_attention_decomposition(query, key, value, attention_mask, scale, casual);
+        model_ref = std::make_shared<ov::Model>(NodeVector{scaled_dot_product_attention},
+                                                ParameterVector{query, key, value, attention_mask, scale});
+    }
+}
+
```
Comment (mmikolajcz) [inline src/common/transformations/tests/op_conversions/scaled_dot_product_decomposition_test.cpp:93]: Maybe add few more tests that would ensure that supported broadcasts of batches and other corresponding elements work as expected?

File: src/common/transformations/tests/op_conversions/scaled_dot_product_decomposition_test.cpp (hunk 4 above)
Comment (PiotrKrzem) [reply src/common/transformations/tests/op_conversions/scaled_dot_product_decomposition_test.cpp]: Right, changed to match the specs

File: src/common/transformations/tests/op_conversions/scaled_dot_product_decomposition_test.cpp (hunk 5 above)
Comment (PiotrKrzem) [reply src/common/transformations/tests/op_conversions/scaled_dot_product_decomposition_test.cpp:161]: Added 2 more tests with crazy shapes to verify Broadcast feasibility

**Review by mmikolajcz — APPROVED** [review]

**Pull request merged**